drivers/adc/adm1177/adm1177.c

```diff
@@ -0,0 +1,274 @@
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+******************************************************************************/
+#include <stdlib.h>
+#include "adm1177.h"
+#include "no_os_alloc.h"
+#include "no_os_delay.h"
+#include "no_os_error.h"
+
+/**************************************************************************//**
+ * @brief Initializes I2C.
+ *
+ * @param device - The device structure.
+ * @param init_param - The structure that contains the device initial
+ * 		       parameters.
+ *
+ * @return status       - Initialization status.
+ *                        Example:  negative value - Initialization failed.
+ *                                  0 - Initialization succeeded.
+******************************************************************************/
+int adm1177_init(struct adm1177_dev **device,
+		 struct adm1177_init_param *init_param)
+{
+	struct adm1177_dev *dev;
+	int status;
+
+	dev = (struct adm1177_dev *)no_os_calloc(sizeof(*dev), 1);
+	if(!dev)
+		return -ENOMEM;
+
+	/* Initialize I2C peripheral */
+	status = no_os_i2c_init(&dev->i2c_desc, &init_param->i2c_init);
+	if(status) {
+		no_os_free(dev);
+		return status;
+	}
+
+	*device = dev;
+
+	return status;
+}
+
+/**************************************************************************//**
+ * @brief Free the resources allocated by adm1177_init().
+ *
+ * @param device - The device structure.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+******************************************************************************/
+int adm1177_remove(struct adm1177_dev *device)
+{
+	int ret;
+
+	ret = no_os_i2c_remove(device->i2c_desc);
+
+	no_os_free(device);
+
+	device = NULL;
```

Comment:
please remove. This is redundant in the way we pass pointers in the `_remove` function.

Reply:
Will remove.